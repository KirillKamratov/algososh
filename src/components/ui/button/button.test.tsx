import React from 'react';
import { Button } from "./button";
import { render, fireEvent, screen } from "@testing-library/react";
import "@testing-library/jest-dom/extend-expect";
import renderer from "react-test-renderer";

const text = 'Test passed'

describe('Testing button component', () => {
  it('button with text', () => {
    const tree = renderer.create( <Button text={text} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('button without text', () => {
    const tree = renderer.create( <Button />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('inactive button', () => {
    const tree = renderer.create( <Button disabled={true} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('loader on button', () => {
    const tree = renderer.create( <Button isLoader={true} />).toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('Onclick is working correctly', () => {
    const onClick = jest.fn()
    render(<Button onClick={onClick} text={text}/>)
    fireEvent.click(screen.getByRole('button'))
    expect(onClick).toHaveBeenCalled()
  })
})

