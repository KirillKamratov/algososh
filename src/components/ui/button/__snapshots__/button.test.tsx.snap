// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing button component button with text 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  >
    Test passed
  </p>
</button>
`;

exports[`Testing button component button without text 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Testing button component inactive button 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  disabled={true}
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Testing button component loader on button 1`] = `
<button
  className="text text_type_button text_color_primary button undefined loader "
  disabled={true}
  type="button"
>
  <img
    alt="Загрузка."
    className="loader_icon"
    src="loader.svg"
  />
</button>
`;
